src/Interceptor/Impl/ListenInterceptor.php

```diff
@@ -33,6 +37,21 @@ public function startUp(Instance $instance): Response
         return new Response();
     }
 
+    private function isInfinityLoop(string $eventName, string $className): bool
+    {
+        $pattern = '#^('.\implode('|', \array_column(Moment::cases(), 'value')).')\.#';
+        if (\preg_match($pattern, $eventName) === 1) {
+            $eventNameWithoutPrefix = \explode('.', $eventName)[1];
+        } else {
+            $eventNameWithoutPrefix = \explode('.', $eventName)[0];
+        }
```

Comment:
it's probably simpler if we use preg_replace here (pre. by nothing) instead if using if